### PR TITLE
Reth primitives traits tx type

### LINKED ISSUE #11531: Define trait `TxType`
### Describe the feature

Define trait `reth_primitives_traits::TxType`. This trait should implement behaviour of `reth_primitives::TxType` and hence  doesn't seem to need any trait methods. It should take as super traits all the `From`/`TryFrom` impls for `TxType` (and those `From<TxType>`/`TryFrom<TxType>` impls that translate to `Into<T>`/`TryInto<T>` impls). Additionally it should take as super traits all the trait bounds on `alloy_consensus::Network::TxType

https://github.com/alloy-rs/alloy/blob/48be08ff0bea3ce49f09ac2d60aaaa3c850dda0b/crates/network/src/lib.rs#L39-L54

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by 0xriazaka**
Closes #11531 
Define trait reth_primitives_traits::TxType

### CODE REVIEW

File: crates/primitives-traits/src/tx_type.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+use core::fmt::{Debug, Display};
+
+/// Trait representing the behavior of a transaction type.
+pub trait TxType:
+    Into<u8>
+    + Into<U8>
+    + PartialEq
+    + Eq
+    + PartialEq<u8>
+    + TryFrom<u8, Error = Eip2718Error>
```
Comment (Rjected) [inline crates/primitives-traits/src/tx_type.rs:13]: do we need the `Error = ` specification here?

File: crates/primitives-traits/src/tx_type.rs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    + TryFrom<u64>
+    + TryFrom<U64>
+    + From<alloy_consensus::TxType>
```
Comment (Rjected) [inline crates/primitives-traits/src/tx_type.rs:16]: is this `from` required?

**Review by Rjected — CHANGES_REQUESTED** [review]
I have a few questions, adding an empty trait seems fine until we finally integrate, where we might need to add some methods

File: crates/primitives-traits/src/tx_type.rs (hunk 1 above)
Comment (0xriazaka) [reply crates/primitives-traits/src/tx_type.rs:13]: yes

File: crates/primitives-traits/src/tx_type.rs (hunk 2 above)
Comment (0xriazaka) [reply crates/primitives-traits/src/tx_type.rs:16]: #11531 yes

**Comment by emhane** [comment]
need help with this @0xriazaka ? it's the key pr for making progress on sdk work rn

**Comment by 0xriazaka** [comment]
> need help with this @0xriazaka ? it's the key pr for making progress on sdk work rn

check the pr

**Review by Rjected — APPROVED** [review]
this looks good to me

**Pull request merged**